From c9e7711ebeba7e8f989e0ffe9a340d3b8319560d Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Wed, 5 Feb 2014 23:02:17 -0800
Subject: [PATCH 026/201] Camera: Fix saturation, contrast, sharpness
 parameters

* Old-style QCOM HALs use <setting>-max, new style HALs use
  max-<setting> for these parameters.

Change-Id: I3b0dbd04f072cb69d09c44b5b3af1097560205b3
---
 src/com/android/camera/CameraSettings.java | 6 +++---
 src/com/android/camera/PhotoModule.java    | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index fb4ddf758..9074ded82 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -848,17 +848,17 @@ public class CameraSettings {
         }
 
         if (saturation != null && !CameraUtil.isSupported(mParameters, "saturation") &&
-                !CameraUtil.isSupported(mParameters, "saturation-max")) {
+                !CameraUtil.isSupported(mParameters, "max-saturation")) {
             removePreference(group, saturation.getKey());
         }
 
         if (contrast != null && !CameraUtil.isSupported(mParameters, "contrast") &&
-                !CameraUtil.isSupported(mParameters, "contrast-max")) {
+                !CameraUtil.isSupported(mParameters, "max-contrast")) {
             removePreference(group, contrast.getKey());
         }
 
         if (sharpness != null && !CameraUtil.isSupported(mParameters, "sharpness") &&
-                !CameraUtil.isSupported(mParameters, "sharpness-max")) {
+                !CameraUtil.isSupported(mParameters, "max-sharpness")) {
             removePreference(group, sharpness.getKey());
         }
 
diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 09a6e97cf..d015ccfa5 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -3191,7 +3191,7 @@ public class PhotoModule
 
         //Set Saturation
         if (CameraUtil.isSupported(mParameters, "saturation") &&
-                CameraUtil.isSupported(mParameters, "saturation-max")) {
+                CameraUtil.isSupported(mParameters, "max-saturation")) {
             String saturationStr = mPreferences.getString(
                     CameraSettings.KEY_SATURATION,
                     mActivity.getString(R.string.pref_camera_saturation_default));
@@ -3203,7 +3203,7 @@ public class PhotoModule
         }
         // Set contrast parameter.
         if (CameraUtil.isSupported(mParameters, "contrast") &&
-                CameraUtil.isSupported(mParameters, "contrast-max")) {
+                CameraUtil.isSupported(mParameters, "max-contrast")) {
             String contrastStr = mPreferences.getString(
                     CameraSettings.KEY_CONTRAST,
                     mActivity.getString(R.string.pref_camera_contrast_default));
@@ -3215,7 +3215,7 @@ public class PhotoModule
         }
         // Set sharpness parameter
         if (CameraUtil.isSupported(mParameters, "sharpness") &&
-                CameraUtil.isSupported(mParameters, "sharpness-max")) {
+                CameraUtil.isSupported(mParameters, "max-sharpness")) {
             String sharpnessStr = mPreferences.getString(
                     CameraSettings.KEY_SHARPNESS,
                     mActivity.getString(R.string.pref_camera_sharpness_default));
-- 
2.17.1

